test/common/testlib.py

```diff
@@ -516,59 +516,33 @@ def blur(self, selector: str) -> None:
+            self.cdp.invoke("Input.dispatchKeyEvent", type="keyUp", text=char, key=char)
+
+    def key(self, name: str, repeat: int = 1, modifiers: int = 0) -> None:
+        args: dict[str, int | str] = {}
         if self.cdp.browser.name == "chromium":
-            self._key_press_chromium(keys, modifiers, use_ord)
-        else:
-            self._key_press_firefox(keys, modifiers, use_ord)
-
-    def _key_press_chromium(self, keys: str, modifiers: int = 0, use_ord: bool = False) -> None:
-        for key in keys:
-            args = {"type": "keyDown", "modifiers": modifiers}
-
-            # If modifiers are used we need to pass windowsVirtualKeyCode which is
-            # basically the asci decimal representation of the key
-            args["text"] = key
-            if use_ord:
-                args["windowsVirtualKeyCode"] = ord(key)
-            elif (not key.isalnum() and ord(key) < 32) or modifiers != 0:
-                args["windowsVirtualKeyCode"] = ord(key.upper())
-            else:
-                args["key"] = key
-
-            self.cdp.invoke("Input.dispatchKeyEvent", **args)
-            args["type"] = "keyUp"
-            self.cdp.invoke("Input.dispatchKeyEvent", **args)
-
-    def _key_press_firefox(self, keys: str, modifiers: int = 0, use_ord: bool = False) -> None:
-        # https://python-reference.readthedocs.io/en/latest/docs/str/ASCII.html
-        # Both line feed and carriage return are normalized to Enter (https://html.spec.whatwg.org/multipage/form-elements.html)
-        keyMap = {
-            8: "Backspace",   # Backspace key
-            9: "Tab",         # Tab key
-            10: "Enter",      # Enter key (normalized from line feed)
-            13: "Enter",      # Enter key (normalized from carriage return)
-            27: "Escape",     # Escape key
-            37: "ArrowLeft",  # Arrow key left
-            40: "ArrowDown",  # Arrow key down
-            45: "Insert",     # Insert key
-        }
-        for key in keys:
-            args = {"type": "keyDown", "modifiers": modifiers}
-
-            args["key"] = key
-            if ord(key) < 32 or use_ord:
-                args["key"] = keyMap[ord(key)]
-
-            self.cdp.invoke("Input.dispatchKeyEvent", **args)
-            args["type"] = "keyUp"
-            self.cdp.invoke("Input.dispatchKeyEvent", **args)
-
-    # FIXME: This should be key_press(), and key_press() should be something else
-    def key(self, name: str) -> None:
-        self.cdp.invoke("Input.dispatchKeyEvent", type="keyDown", key=name)
-        self.cdp.invoke("Input.dispatchKeyEvent", type="keyUp", key=name)
+            # HACK: chromium doesn't understand some key codes in some situations
+            win_keys = {
+                "Backspace": 8,
+                "Enter": 13,
+                "Escape": 27,
+                "ArrowDown": 40,
+                "Insert": 45,
+            }
```

Comment:
Interesting, its a bit different from https://github.com/cockpit-project/cockpit-files/blob/main/test/check-application#L47

So it would be nice to add ArrowUp/ArrowLeft/ArrowRight if that is still needed

Reply:
That's what I hinted at in the commit message: c-storage-partitions already uses ArrowLeft and that seems to be fine. I don't see any pattern here yet. But I expect some more fallout from this indeed, we may have to adjust that a bit in the future still.

Reply:
I locally tested this branch against https://github.com/cockpit-project/cockpit-files/pull/543 and it all passed.